### PR TITLE
add new case for luks raid

### PR DESCRIPTION
**Pull request by guazhangRH**
this PR is same with PR130, just switch git branch and submit again.

add case for luks vol on sw raid device.

### CODE REVIEW

File: tests/tests_luks_raid.yml [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+            name: storage
+          vars:
+            safe_mode: false
+            storage_volumes:
+              - name: foo
+                type: raid
+                disks: "{{ unused_disks }}"
+                raid_level: "raid0"
+                encryption: true
+                volumes:
```
Comment (pcahyna) [inline tests/tests_luks_raid.yml:39]: How can a storage volume have other volumes under it? Shouldn't it be under `storage_pools`, not `storage_volumes`? @japokorn @dwlehman

Comment (japokorn) [reply tests/tests_luks_raid.yml:39]: PR description mentions volumes so I presume that `storage_volumes` should be used here. `storage_volumes` does not support `volumes` attribute though.
I think that what is happening in this case is that `volumes` makes it into the `storage_volumes` dictionary w/o producing an error but then it is never used.

To fix this, the `volumes` contents should be moved directly into `storage_volumes` dictionary:
```
storage_volumes:
  - name: test1
    type: raid
    disks: "{{ unused_disks }}"
    raid_level: "raid0"
    encryption: true
    mount_point: "{{ mount_location }}"
    size: 4g
```

Comment (dwlehman) [reply tests/tests_luks_raid.yml:39]: Good catch. This is repeated below as well.

Comment (pcahyna) [reply tests/tests_luks_raid.yml:39]: If it is not used, something is wrong with the result validation in test.

Comment (guazhangRH) [reply tests/tests_luks_raid.yml]: > If it is not used, something is wrong with the result validation in test.
Thanks for the review, I have fixed the error. 
BTW, I don't know why it's works before and test pass

File: tests/tests_luks_raid.yml [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    storage_safe_mode: false
+    mount_location: '/opt/test1'
+    storage_use_partitions: true
+    volume_size: '5g'
+    key_file: '/opt/keyfile'
+    passwd: 'yabbadabbadoo'
+
+  tasks:
+    - include_role:
+        name: storage
```
Comment (richm) [inline tests/tests_luks_raid.yml:14]: please use `name: linux-system-roles.storage`

File: tests/tests_luks_raid.yml [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+    ##
+    ## Raid
+    ##
+
+    - name: Test for correct handling of new encrypted pool w/ no key
+      block:
+        - name: Create an encrypted raid volume
+          include_role:
+            name: storage
```
Comment (richm) [inline tests/tests_luks_raid.yml:30]: please use `name: linux-system-roles.storage`

File: tests/tests_luks_raid.yml [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+        - name: Verify the output of the keyless luks test
+          assert:
+            that: "blivet_output.failed and
+                   blivet_output.msg|regex_search('encrypted volume.*missing key') and
+                   not blivet_output.changed"
+            msg: "Unexpected behavior w/ encrypted raid pool w/ no key"
+
+    - name: Create an encrypted raid volume w/ default fs
+      include_role:
+        name: storage
```
Comment (richm) [inline tests/tests_luks_raid.yml:61]: please use `name: linux-system-roles.storage`

File: tests/tests_luks_raid.yml [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+            encryption_key_size: 512
+            encryption_luks_version: 'luks2'
+            mount_point: "{{ mount_location }}"
+            size: 4g
+
+    - include_tasks: verify-role-results.yml
+
+    - name: Remove the encryption layer
+      include_role:
+        name: storage
```
Comment (richm) [inline tests/tests_luks_raid.yml:80]: please use `name: linux-system-roles.storage`

File: tests/tests_luks_raid.yml [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+    - include_tasks: verify-role-results.yml
+
+    - name: Generate a keyfile as password
+      copy:
+        dest: "{{ key_file }}"
+        content: 'redhat redhat'
+        
+    - name: Add encryption to the volume
+      include_role:
+        name: storage
```
Comment (richm) [inline tests/tests_luks_raid.yml:101]: please use `name: linux-system-roles.storage`

File: tests/tests_luks_raid.yml [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+    - include_tasks: verify-role-results.yml
+
+    - name: Remove keyfile
+      file:
+        path: "{{ key_file }}"
+        state: absent
+
+    - name: Clean up
+      include_role:
+        name: storage
```
Comment (richm) [inline tests/tests_luks_raid.yml:125]: please use `name: linux-system-roles.storage`

Comment (guazhangRH) [reply tests/tests_luks_raid.yml]: Hi,

have rename storage to linux-system-roles.storage, Please review

Comment (richm) [reply tests/tests_luks_raid.yml]: > have rename storage to linux-system-roles.storage, Please review

lgtm

**Comment by guazhangRH** [comment]
Could someone help to merged the PR if the code looks good

**Comment by richm** [comment]
[citest pending]

**Comment by richm** [comment]
[citest bad]

**Comment by richm** [comment]
[citest pending]

**Comment by richm** [comment]
[citest pending]

**Comment by richm** [comment]
Do we still need this?

**Comment by richm** [comment]
[citest pending]

**Pull request closed without merging**